output/eu_crude/analysis.py

```diff
@@ -74,8 +78,148 @@ def get_waste_id(resource_array):
+                              WHERE spec = ":cycamore:DeployInst"').fetchall()
+
+    entry = cur.execute('sELECT power_cap, agententry.agentid, parentid, entertime\
+                        FROM agententry INNER JOIN\
+                        agentstate_cycamore_reactorinfo\
+                        ON agententry.agentid =\
+                        agentstate_cycamore_reactorinfo.agentid').fetchall()
+
+    exit = cur.execute('sELECT power_cap, agentexit.agentid, parentid, exittime\
+                        FROM agentexit INNER JOIN\
+                        agentstate_cycamore_reactorinfo\
+                        ON agentexit.agentid =\
+                        agentstate_cycamore_reactorinfo.agentid\
+                        INNER JOIN agententry\
+                        ON agentexit.agentid = agententry.agentid').fetchall()
+
+    # create empty array for each country_government
+    for gov in governments:
+        exec(gov[0] + '_power = []', globals())
+        exec(gov[0] + '_num = []', globals())
+        countries.append(gov[0])
+
+    # add and subtract capacity and num for every government
+    # for every timestep for every entry and exit
+    for gov in governments:
+        cap = 0
+        count = 0
+        for num in timestep:
+            for enter in entry:
+                if enter[3] == num and enter[2] == gov[1]:
+                    cap += enter[0]
+                    count += 1
+            for dec in exit:
+                if dec[3] == num and dec[2] == gov[1]:
+                    cap -= dec[0]
+                    count -= 1
+            exec(gov[0] + '_power.append(cap)')
+            exec(gov[0] + '_num.append(count)')
+
+    colores = get_colors()
+
+    # set different colors for each bar
+    index = 0
+    top_index = 0
+    prev_govs = ''
+
+    # plot string for the legend
+    plot_string = 'plt.legend(('
+
+    # convert power arrays to numpy arrays for stacking
+    for gov in governments:
+        exec(gov[0] + '_power = np.asarray(' + gov[0] + '_power)')
+
+    # for every country, create bar chart with different color
+    for gov in governments:
+        color = colores[index]
+        # very first country does not have a 'bottom' argument
+        if top_index == 0:
+            exec(
+                gov[0] + '_plot = plt.bar( 1950+(timestep/12), '
+                + gov[0] + '_power, .5, color = color,\
+                edgecolor = "none")'
+                )
+            prev_govs = gov[0] + '_power'
+        # from the second country has 'bottom' argument
+        else:
+            exec(
+                gov[0] + '_plot = plt.bar( 1950+(timestep/12), '
+                + gov[0] + '_power, .5, color = color,\
+                edgecolor = "none", bottom =' + prev_govs + ')'
+                )
+            prev_govs += '+ ' + gov[0] + '_power'
+
+        plot_string += gov[0] + '_plot,'
+        index += 1
+        top_index += 1
+
+    # generate string for exec of plt.lengend
```

Comment:
Maybe you can explain verbally why this is necessary. Still not clear what you're trying to do.